### PR TITLE
Wallet - implemented wallet_send_basic_outputs

### LINKED ISSUE #381: Wallet - Gather node info and store it for later use
## Description

As a wallet, it should be able to get node info from "/api/v2/info" and store it locally for using in other methods. This can be done during wallet create.

### LINKED ISSUE #383: Wallet - Collect needed inputs for a transaction
## Description

A method that fetches and selects UTXO_Inputs needed for the transaction outputs.

### PR DESCRIPTION
**Pull request by oopsmonk**
# Description of change

- fixes #381 
- fixes #383 

## Type of change

Choose a type of change, and delete any options that are not relevant.

- Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

tested token transfer with a private Tangle

Make sure to provide instructions for the maintainer as well as any relevant configurations.

## Change checklist

Add an `x` to the boxes that are relevant to your changes, and delete any items that are not.

- [x] My code follows the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests using CTest that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/core/models/outputs/byte_cost_config.c [hunk 1]
```diff
@@ -40,3 +40,16 @@ void byte_cost_config_free(byte_cost_config_t *config) {
     free(config);
   }
 }
+
+void byte_cost_config_set(byte_cost_config_t *config, uint16_t byte_cost, uint8_t byte_factor_data,
+                          uint8_t byte_factor_key) {
```
Comment (matjazv) [inline src/core/models/outputs/byte_cost_config.c:45]: `config` could be NULL. Maybe a NULL check could be added.

File: src/wallet/wallet.c [hunk 2]
```diff
@@ -92,128 +96,152 @@ static void get_address_path(uint32_t account, bool change, uint32_t index, char
...
-  if (outputs_res->is_error) {
-    printf("[%s:%d] Error get outputs from addr: %s\n", __func__, __LINE__, outputs_res->u.error->msg);
-    goto done;
-  }
+  if (address_to_bech32(send_addr, w->bech32HRP, bech32_addr, sizeof(bech32_addr)) == 0) {
+    outputs_query_list_t* query_param = outputs_query_list_new();
+    ret = outputs_query_list_add(&query_param, QUERY_PARAM_ADDRESS, bech32_addr);
+    if (ret != 0) {
+      printf("[%s:%d] add query params failed\n", __func__, __LINE__);
+      outputs_query_list_free(query_param);
```
Comment (matjazv) [inline src/wallet/wallet.c:118]: `res_id` should also be freed.

```suggestion
      outputs_query_list_free(query_param);
      res_outputs_free(res_id);
```

File: src/wallet/wallet.c [hunk 3]
```diff
@@ -347,7 +375,7 @@ void wallet_destroy(iota_wallet_t* w) {
   }
 }
 
-int wallet_update_bech32HRP(iota_wallet_t* w) {
+int wallet_update_node_config(iota_wallet_t* w) {
```
Comment (matjazv) [inline src/wallet/wallet.c:379]: `w` could be `NULL`. Maybe a `NULL` check could be added.

File: src/wallet/wallet.c [hunk 4]
```diff
@@ -366,3 +412,114 @@ int wallet_update_bech32HRP(iota_wallet_t* w) {
   res_node_info_free(info);
   return ret;
 }
+
+int wallet_send_basic_outputs(iota_wallet_t* w, bool change, uint32_t index, address_t* recv_addr, uint64_t send_amount,
```
Comment (matjazv) [inline src/wallet/wallet.c:422]: `w` and `recv_addr` could be `NULL`. Maybe a `NULL` check could be added.

File: tests/wallet/test_wallet.c [hunk 5]
```diff
@@ -1,10 +1,27 @@
...
+
+// API endpoint
+#define NODE_HOST "localhost"
+#define NODE_PORT 14265
+#define NODE_IS_HTTPS 0
+
+// faucet endpoint
+#define FAUCET_HOST "localhost"
+#define FAUCET_PORT 8091
+#define FAUCET_IS_HTTPS 0
```
Comment (matjazv) [inline tests/wallet/test_wallet.c:23]: Should a `wallet` module also have a common `test_config.h` file? Or maybe whole library could have a common one?

**Review by matjazv — CHANGES_REQUESTED** [review]

File: tests/wallet/test_wallet.c (hunk 5 above)
Comment (oopsmonk) [reply tests/wallet/test_wallet.c:23]: yes, those configurations are temporary, we will have a common configuration and flags for online, local test scenarios with the node.

**Review by matjazv — APPROVED** [review]

**Review by rebinmannady93 — APPROVED** [review]

**Pull request merged**